File: java/third.java
import java.util.ArrayList;

public class third {
    public static void main(String[] args) {
        long number = 600851475143L;
        ArrayList<Integer> largePrime = new ArrayList<Integer>();
        largePrime.add(2);
        boolean prime = true;

        for(int i = 3; i < (number/2); i++)
            if(number % i == 0){
                for(int j : largePrime){
                    if(i % j == 0){
                        prime = false;
                        break;
                    }
                    if(prime)
                        largePrime.add(i);
                }
            }
        
        System.out.println(largePrime.get(largePrime.size()-1));
    }
}

File: python/third.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jan 20 14:41:53 2023

@author: Amir
"""

number = 600851475143 
largePrime = [2]
prime = True

for i in range(3, int(number/2)):
    if number % i == 0:
        for j in largePrime:
            if i % j == 0:
                prime = False
                break
        if prime:
            largePrime.append(i)
            
        
print(largePrime)